### ISSUE TITLE
Add assume role ARN in the EC2NodeClass

### ISSUE THREAD

**Issue by mcanevet** [aws/karpenter#5233]
### Description

**What problem are you trying to solve?**

It looks like it's possible to leverage Karpenter to create a node in a different account than the account hosting the control plane using [shared VPC architecture](https://github.com/aws/karpenter/pull/1732), if so, I guess it uses the IAM identity of the controller to create the EC2 instance, and thus create it in the same account than the IAM role, right? So I guess it supports only one account per controller. If so, I could deploy one controller per account in which I'd like to deploy an EC2, but in that case how can I tell a specific controller to manage only some NodePools? Maybe an elegant solution would be to have an assume role attribute in the EC2NodeClass so that we could have only one controller that would switch to a role in another account depending on the NodePool that has been selected to create the node?

**How important is this feature to you?**

This would allow us to have EC2 in different accounts.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by sftim** [comment]
Some [context](https://kubernetes.slack.com/archives/C02SFFZSA2K/p1701779290329679)

**Comment by tzneal** [comment]
There is a feature request at https://github.com/aws/containers-roadmap/issues/307 to support nodes in a different account in EKS.  Until that is supported, this won't be possible.  

I'll go ahead and close this one as a duplicate of that issue.

**Comment by sftim** [comment]
@tzneal, this feature is useful for people who use Karpenter but not EKS.

**Comment by tzneal** [comment]
I didn't consider that case, I'll re-open it.

**Comment by jonathan-innis** [comment]
It's currently possible to do this for a single account with `--assume-role-arn` set on the controller, but (like you mentioned) this doesn't allow you to do multi-account, only provision instances in a different, single account.

**Comment by mcanevet** [comment]
@jonathan-innis Yes, that's what I saw in the documentation. And being able to run one Karpenter controller per account would do the trick if we where able to monitor only some NodePools for each controller, but it looks like it's not possible. Anyway, having a single controller and being able to assume a different role per EC2NodeClass would be better IMHO. So we need to store the information of the role to assume at EC2NodeClass level, not controller level. Plus it would make the controller more cloud agnostic?

**Comment by tuananh** [comment]
This would be a very nice addition.
